### DISCUSSION TITLE
Shell completion

### DISCUSSION THREAD

**Discussion by mickael-menu** [zk-org/zk#615] [Ideas]
### Discussed in https://github.com/mickael-menu/zk/discussions/165

<div type='discussions-op-text'>

<sup>Originally posted by **rgoncalvesse** February  4, 2022</sup>

Am I am missing something or is there still no shell completion?

Actually, I would separate the completion in two distinct types:

1. Basic completion. Suggesting commands (e.g. new, list, ...) and flags for each of those commands (e.g. --help, -f/--format, ...).
2. Filter completion. Suggesting note title, note id or tags when filtering (e.g. https://github.com/mickael-menu/zk/blob/main/docs/note-filtering.md#filter-by-tags)

For instance, `taskwarrior` suggests a multitude of dates/day/month/events when creating a new task with a due date. For zk, the ultimate completion state could be to suggest similar days/events when filtering notes by creation or modification (see below).

![image](https://user-images.githubusercontent.com/18457507/152557882-4562b5c8-a7ee-4ba6-b97a-55598ea71dc4.png)
</div>

**Comment by dandeandean** [comment]
Do you guys know if any work has been done on this? I think even the simple completion would be a huge win for cli usability. It'd be nice to be able to page through the notes by calling `zk edit <t>`. If no one is actively working on it, I could take a look.

**Comment by tjex** [comment]
This is not being worked on since the start of 2024 when I came on board. Before that, I'm not sure, but that's more or less a general "no" I'd say. 

I agree, it would be nice :)

**Comment by WhyNotHugo** [comment]
> It'd be nice to be able to page through the notes by calling zk edit <t>.

`zk edit -i` should work for that particular use case.

**Comment by GV14982** [comment]
Any interest in this being worked on? I'd be happy to contribute!
